manifests/init.pp

```diff
@@ -34,215 +34,228 @@
 #   String.  Specifies if jdk should be installed or absent
 #   Defaults to <tt>installed</tt>.
 #
-class jdk_oracle(
-  $version      = hiera('jdk_oracle::version',        '8' ),
-  $arch         = hiera('jdk_oracle::arch',           'x64' ),
-  $install_dir  = hiera('jdk_oracle::install_dir',    '/usr/java' ),
-  $tmp_dir      = hiera('jdk_oracle::tmp_dir',        '/tmp' ),
-  $use_cache    = hiera('jdk_oracle::use_cache',      false ),
-  $cache_source = 'puppet:///modules/jdk_oracle/',
-  $jce          = hiera('jdk_oracle::jce',            false ),
-  $default_java = hiera('jdk_oracle::default_java',   true ),
-) {
-
-  # Set default exec path for this module
-  Exec { path    => ['/usr/bin', '/usr/sbin', '/bin'] }
-
-  case $version {
-    '7': {
-      $javaUpdate = hiera('jdk_oracle::version::7::update', '80')
-      $javaBuild = hiera('jdk_oracle::version::7::build',   '15')
-    }
-    '8': {
-      $javaUpdate = hiera('jdk_oracle::version::8::update', '91')
-      $javaBuild = hiera('jdk_oracle::version::8::build',   '14')
-    }
-    default: {
-      fail("Unsupported version: ${version}.  Implement me?")
-    }
-  }
-
-  $java_home = "${install_dir}/jdk1.${version}.0_${javaUpdate}"
-  $javaDownloadURI = "http://download.oracle.com/otn-pub/java/jdk/${version}u${javaUpdate}-b${javaBuild}/jdk-${version}u${javaUpdate}-linux-${arch}.rpm"
-  $jceDownloadURI = 'http://download.oracle.com/otn-pub/java/jce/8/jce_policy-8.zip'
+class jdk_oracle (
+    $version      = 8,
```

Comment:
Additionally, wouldn't this mean anyone who has explicitly set the version would have to perform a change as their configuration would no longer be compatible?